packages/shared/src/crypto.ts

```diff
@@ -91,22 +105,27 @@ export const generateKey = (
     );
 
     // Hash and Encode the parts and apiKey as hex strings
-    const encodedFileSeed = yield* sha256Hex(hashParts);
-    const encodedApiKey = yield* sha256Hex(apiKey);
+    const alphabet = shuffle(defaultOptions.alphabet, apiKey);
+    const encodedFileSeed = new SQIds({ alphabet, minLength: 36 }).encode([
+      Math.abs(Hash.string(hashParts)),
+    ]);
+    const encodedApiKey = new SQIds({ alphabet, minLength: 12 }).encode([
+      Math.abs(Hash.string(apiKey)),
+    ]);
 
     // Concatenate them and encode as base64
-    return Encoding.encodeBase64([encodedApiKey, encodedFileSeed].join(":"));
+    return [encodedApiKey, encodedFileSeed].join(KEY_SEPARATOR);
```

Comment:
Is this URL safe? I chose it cause it "looks" like a normal character so it doesn't stand out as much xd

could also base64 encode the key, but it does add to it's length

```sh
KJNGYKlZPq92|4wGwXUsC5WekfPrntiTE0qUyIo91O87zXwNg => 49ch
S0pOR1lLbFpQcTkyfDR3R3dYVXNDNVdla2ZQcm50aVRFMHFVeUlvOTFPODd6WHdOZw== 68ch
```

Reply:
Pipe is not urlsafe, could use dash?

Reply:
do they need to be separated?

Reply:
i guess not really if we do a `startsWith` check instead.

I think the reason for the separation initially was adding a timing safe equals but that might be overkill?

Reply:
Yeah, I think it's fine if it isn't timing safe